runtime/block_reduction.cu

```diff
@@ -15,46 +15,38 @@
 //  blockReduceSum<X_THREADS, Y_THREADS, Z_THREADS>
 //    (output[output_index], inputs[input_index],
 //      [] __device__ (T& a, const T b) { a += b; });
-//
-// Note: We agressively template functions taking dim3 in the functions below
-//       because ROCM uses different types for the various dim3 and maps them
-//       directly to intrinsics, but they're dim3 when used after modification.
-//
 template <
     bool X_REDUCE,
     bool Y_REDUCE,
     bool Z_REDUCE,
+    bool aligned,
     typename T,
-    typename Func,
-    typename _dim3,
-    typename _dim3_2>
```

Comment:
Not necessary anymore.

Reply:
Was this only for ROCm? This looks annoying and make the code unreadable. It's good that we can remove it now.

Reply:
Yes, it was for the compatibility